solutions/devsprint-pedro-menezes-1/DeliveryAppChallenge/Modules/Product/Scenes/PedroTres/Login/PTLoginViewController.swift

```diff
@@ -108,50 +85,75 @@ class PTLoginViewController: UIViewController {
         }
     }
     
-    private func handleSuccess(session: Session) {
+    func showAlert() {
+        let alertController = UIAlertController(title: "Sem conexão", message: "Conecte-se à internet para tentar novamente", preferredStyle: .alert)
+        let actin = UIAlertAction(title: "Ok", style: .default)
+        alertController.addAction(actin)
+        present(alertController, animated: true)
+    }
+    
+    func navigateToHome(){
+        let vc = UINavigationController(rootViewController: HomeViewController())
+        let scenes = UIApplication.shared.connectedScenes
+        let windowScene = scenes.first as? UIWindowScene
+        let window = windowScene?.windows.first
+        window?.rootViewController = vc
+        window?.makeKeyAndVisible()
+    }
+    
+    func isValidEmail(_ email: String?) -> Bool {
+        guard let email = email else {
+            return false
+        }
+        return !email.contains(".") ||
+        !email.contains("@") ||
+        email.count <= 5
+    }
+    
+    func handleSuccess(session: Session) {
```

Comment:
handle success de q?
falta uma palavra para esse nome ser mais descritivo e de acordo com seu contexto
sugiro tb user with, of, from antes do nome do parametro
aqui faria sentido `(with session: Session)`